### PR TITLE
State layout

### PR DESCRIPTION
**Pull request by MayuBen1027**
Adding the layout state based on the info found [here](https://github.com/inkdropapp/docs/blob/master/src/pages/reference/state-layouts.md?plain=1).

### CODE REVIEW

File: src/app/states/layouts/page.mdx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+## Data Structure
+A map of `[layout_name]: LayoutComponents` where:
+
+- `layout_name`: String
+- `LayoutComponents`: A class name of React component
+
+<Row>
+  <Col>
+    <Properties>
+      <Property name="main" type="full">
```
Comment (Keisir) [inline src/app/states/layouts/page.mdx:23]:
```suggestion
      <Property name="main:full" type="Array<string>">
```
This should make it easier to understand.
The `:` is part of the layout name and therefore it should also be directly in the name

File: src/app/states/layouts/page.mdx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+# layouts
+
+An [app state](/guides/flux-architecture) of the layouts.
+You can obtain the `layouts` state and integrate it with React components.{{ className: 'lead' }}
+
+
+---
+
+## Data Structure
+A map of `[layout_name]: LayoutComponents` where:
```
Comment (Keisir) [inline src/app/states/layouts/page.mdx:15]: `Record<string, Array<string>>` makes this more readable I think.
(or `Record<string, LayoutComponents>`)

**Review by Keisir — COMMENTED** [review]
Looks good so far.
To transfer the page completely a few layouts are still missing.

File: src/app/states/layouts/page.mdx (hunk 2 above)
Comment (MayuBen1027) [reply src/app/states/layouts/page.mdx]: Good call out.

File: src/app/states/layouts/page.mdx [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+        ```json
+        []
+        ```
+      </Property>
+      <Property name="editor-floating-actions" type="Array<string>">
+      The floating action buttons on the editor
+
+      **Default values:**
+        ```json
+        ["EditorFloatingActionTogglePreview", "EditorFloatingActionToggleSideBySide"]
```
Comment (Keisir) [inline src/app/states/layouts/page.mdx:156]:
```suggestion
        [
          "EditorFloatingActionTogglePreview",
          "EditorFloatingActionToggleSideBySide"
        ]
```
Better readability by eliminating the need to scroll horizontally

File: src/app/states/layouts/page.mdx [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+        ```json
+        []
+        ```
+      </Property>
+      <Property name="editor-drawer-menu" type="Array<string>">
+      The editor drawer menu items
+
+      **Default values:**
+        ```json
+        ["EditorDrawerSectionHeaderNoteInformation", "EditorDrawerMenuItemNotebook", "EditorDrawerItemStatus", "EditorDrawerItemTags","EditorDrawerItemCreatedDate", "EditorDrawerItemUpdatedDate", "EditorDrawerSeparator-1", "EditorDrawerSectionHeaderActions","EditorDrawerActionButtonDuplicate", "EditorDrawerActionButtonCopyNoteLink", "EditorDrawerActionButtonPinToTop","EditorDrawerActionButtonOpenNoteInSeparateWindow", "EditorDrawerSeparator-2", "EditorDrawerActionButtonDeleteNote","EditorDrawerActionButtonRevisionHistory", "EditorDrawerSeparator-3", "EditorDrawerActionButtonShare"]
```
Comment (Keisir) [inline src/app/states/layouts/page.mdx:140]:
```suggestion
        [
          "EditorDrawerSectionHeaderNoteInformation",
          "EditorDrawerMenuItemNotebook",
          "EditorDrawerItemStatus",
          "EditorDrawerItemTags",
          "EditorDrawerItemCreatedDate",
          "EditorDrawerItemUpdatedDate",
          "EditorDrawerSeparator-1",
          "EditorDrawerSectionHeaderActions",
          "EditorDrawerActionButtonDuplicate",
          "EditorDrawerActionButtonCopyNoteLink",
          "EditorDrawerActionButtonPinToTop",
          "EditorDrawerActionButtonOpenNoteInSeparateWindow",
          "EditorDrawerSeparator-2",
          "EditorDrawerActionButtonDeleteNote",
          "EditorDrawerActionButtonRevisionHistory",
          "EditorDrawerSeparator-3",
          "EditorDrawerActionButtonShare"
        ]
```

File: src/app/states/layouts/page.mdx [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+        ```json
+        ["EditorMetaNotebook", "EditorMetaStatusPicker", "EditorMetaTags"]
+        ```
+      </Property>
+      <Property name="modal" type="Array<string>">
+      The layout for modal components
+
+      **Default values:**
+        ```json
+        ["NewNotebookDialog", "DeleteNotebookDialog", "RenameNotebookDialog", "TagSettingsDialog", "DeleteTagDialog", "AssistiveErrorDialog", "MoveNotesDialog","MoveNotebookDialog", "EmptyTrashDialog", "ShareNoteDialog", "NoteRevisionsDialog"]
```
Comment (Keisir) [inline src/app/states/layouts/page.mdx:124]:
```suggestion
        [
          "NewNotebookDialog",
          "DeleteNotebookDialog",
          "RenameNotebookDialog",
          "TagSettingsDialog",
          "DeleteTagDialog",
          "AssistiveErrorDialog",
          "MoveNotesDialog",
          "MoveNotebookDialog",
          "EmptyTrashDialog",
          "ShareNoteDialog",
          "NoteRevisionsDialog"
        ]
```

File: src/app/states/layouts/page.mdx [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+        ```json
+        ["EditorHeaderTitleInput", "EditorHeaderMore"]
+        ```
+      </Property>
+      <Property name="editor-meta" type="Array<string>">
+      The meta pane layout of the editor
+
+      **Default values:**
+        ```json
+        ["EditorMetaNotebook", "EditorMetaStatusPicker", "EditorMetaTags"]
```
Comment (Keisir) [inline src/app/states/layouts/page.mdx:116]:
```suggestion
        [
          "EditorMetaNotebook",
          "EditorMetaStatusPicker",
          "EditorMetaTags"
        ]
```

File: src/app/states/layouts/page.mdx [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+        ```json
+        ["EditorDrawer", "EditorFloatingActions", "EditorHeaderLayout", "EditorMetaLayout", "EditorToolbar", "MDELayout", "MDESearchBar", "EditorStatusBarLayout"]
+        ```
+      </Property>
+      <Property name="editor-header" type="Array<string>">
+      The header layout of the editor
+
+      **Default values:**
+        ```json
+        ["EditorHeaderTitleInput", "EditorHeaderMore"]
```
Comment (Keisir) [inline src/app/states/layouts/page.mdx:108]:
```suggestion
        [
          "EditorHeaderTitleInput",
          "EditorHeaderMore"
        ]
```

File: src/app/states/layouts/page.mdx [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+        ```json
+        ["Editor"]
+        ```
+      </Property>
+      <Property name="editor" type="Array<string>">
+      The editor layout
+
+      **Default values:**
+        ```json
+        ["EditorDrawer", "EditorFloatingActions", "EditorHeaderLayout", "EditorMetaLayout", "EditorToolbar", "MDELayout", "MDESearchBar", "EditorStatusBarLayout"]
```
Comment (Keisir) [inline src/app/states/layouts/page.mdx:100]:
```suggestion
        [
          "EditorDrawer",
          "EditorFloatingActions",
          "EditorHeaderLayout",
          "EditorMetaLayout",
          "EditorToolbar",
          "MDELayout",
          "MDESearchBar",
          "EditorStatusBarLayout"
        ]
```

File: src/app/states/layouts/page.mdx [hunk 9]
```diff
@@ -0,0 +1,179 @@
...
+        ```json
+        ["NoteListHeader", "NoteListBarContainer"]
+        ```
+      </Property>
+      <Property name="mde" type="Array<string>">
+      The mde layout
+
+      **Default values:**
+        ```json
+        ["Editor"]
```
Comment (Keisir) [inline src/app/states/layouts/page.mdx:92]:
```suggestion
        [
          "Editor"
        ]
```

File: src/app/states/layouts/page.mdx [hunk 10]
```diff
@@ -0,0 +1,179 @@
...
+        ```json
+        ["NoteListHeader", "NoteListSearchBar", "NoteListBarContainer"]
+        ```
+      </Property>
+      <Property name="note-list-bar:trash" type="Array<string>">
+      The trash menu layout of note list bar
+
+      **Default values:**
+        ```json
+        ["NoteListHeader", "NoteListBarContainer"]
```
Comment (Keisir) [inline src/app/states/layouts/page.mdx:84]:
```suggestion
        [
          "NoteListHeader",
          "NoteListBarContainer"
        ]
```

File: src/app/states/layouts/page.mdx [hunk 11]
```diff
@@ -0,0 +1,179 @@
...
+        ```json
+        ["SidebarWorkspaceMenuTitle", "SideBarWorkspaceMenuAllNotes", "SideBarMenuSectionNotebooks", "SideBarWorkspaceMenuBookList", "SideBarMenuSectionStatus", "SidebarWorkspaceMenuStatusList", "SideBarMenuSectionTags", "SidebarWorkspaceMenuTagList"]
+        ```
+      </Property>
+      <Property name="note-list-bar" type="Array<string>">
+      The menu layout of note list bar
+
+      **Default values:**
+        ```json
+        ["NoteListHeader", "NoteListSearchBar", "NoteListBarContainer"]
```
Comment (Keisir) [inline src/app/states/layouts/page.mdx:76]:
```suggestion
        [
          "NoteListHeader",
          "NoteListSearchBar",
          "NoteListBarContainer"
        ]
```

File: src/app/states/layouts/page.mdx [hunk 12]
```diff
@@ -0,0 +1,179 @@
...
+        ```json
+        ["SideBarMenuItemAllNotes", "SideBarMenuSectionNotebooks", "SideBarMenuBookListRoot", "SideBarMenuItemTrash", "SideBarMenuSectionStatus", "SideBarMenuStatusList", "SideBarMenuSectionTags", "SideBarMenuTagList"]
+        ```
+      </Property>
+      <Property name="sidebar-workspace-menu" type="Array<string>">
+      The menu layout of sidebar workspace menu
+
+      **Default values:**
+        ```json
+        ["SidebarWorkspaceMenuTitle", "SideBarWorkspaceMenuAllNotes", "SideBarMenuSectionNotebooks", "SideBarWorkspaceMenuBookList", "SideBarMenuSectionStatus", "SidebarWorkspaceMenuStatusList", "SideBarMenuSectionTags", "SidebarWorkspaceMenuTagList"]
```
Comment (Keisir) [inline src/app/states/layouts/page.mdx:68]:
```suggestion
        [
          "SidebarWorkspaceMenuTitle",
          "SideBarWorkspaceMenuAllNotes",
          "SideBarMenuSectionNotebooks",
          "SideBarWorkspaceMenuBookList",
          "SideBarMenuSectionStatus",
          "SidebarWorkspaceMenuStatusList",
          "SideBarMenuSectionTags",
          "SidebarWorkspaceMenuTagList"
        ]
```

File: src/app/states/layouts/page.mdx [hunk 13]
```diff
@@ -0,0 +1,179 @@
...
+        ```json
+        ["SideBarHeader", "SideBarNavigation", "SideBarSyncStatusView", "SideBarPluginUpdatesNotificationView"]
+        ```
+      </Property>
+      <Property name="sidebar-menu" type="Array<string>">
+      The menu layout of sidebar
+
+      **Default values:**
+        ```json
+        ["SideBarMenuItemAllNotes", "SideBarMenuSectionNotebooks", "SideBarMenuBookListRoot", "SideBarMenuItemTrash", "SideBarMenuSectionStatus", "SideBarMenuStatusList", "SideBarMenuSectionTags", "SideBarMenuTagList"]
```
Comment (Keisir) [inline src/app/states/layouts/page.mdx:60]:
```suggestion
        [
          "SideBarMenuItemAllNotes",
          "SideBarMenuSectionNotebooks",
          "SideBarMenuBookListRoot",
          "SideBarMenuItemTrash",
          "SideBarMenuSectionStatus",
          "SideBarMenuStatusList",
          "SideBarMenuSectionTags",
          "SideBarMenuTagList"
        ]
```

File: src/app/states/layouts/page.mdx [hunk 14]
```diff
@@ -0,0 +1,179 @@
...
+        ```json
+        ["ModalLayout", "EditorLayout"]
+        ```
+      </Property>
+      <Property name="sidebar" type="Array<string>">
+      The sidebar layout
+
+      **Default values:**
+        ```json
+        ["SideBarHeader", "SideBarNavigation", "SideBarSyncStatusView", "SideBarPluginUpdatesNotificationView"]
```
Comment (Keisir) [inline src/app/states/layouts/page.mdx:52]:
```suggestion
        [
          "SideBarHeader",
          "SideBarNavigation",
          "SideBarSyncStatusView",
          "SideBarPluginUpdatesNotificationView"
        ]
```

File: src/app/states/layouts/page.mdx [hunk 15]
```diff
@@ -0,0 +1,179 @@
...
+        ```json
+        ["ModalLayout", "NoteListBarLayout", "EditorLayout"]
+        ```
+      </Property>
+      <Property name="main:distraction-free" type="Array<string>">
+      The main layout for distraction-free mode.
+
+      **Default values:**
+        ```json
+        ["ModalLayout", "EditorLayout"]
```
Comment (Keisir) [inline src/app/states/layouts/page.mdx:44]:
```suggestion
        [
          "ModalLayout",
          "EditorLayout"
        ]
```

File: src/app/states/layouts/page.mdx [hunk 16]
```diff
@@ -0,0 +1,179 @@
...
+        ```json
+        ["ModalLayout", "SideBarLayout", "NoteListBarLayout", "EditorLayout"]
+        ```
+      </Property>
+      <Property name="main:slim" type="Array<string>">
+      The main layout that consists of 2 columns.
+
+      **Default values:**
+        ```json
+        ["ModalLayout", "NoteListBarLayout", "EditorLayout"]
```
Comment (Keisir) [inline src/app/states/layouts/page.mdx:36]:
```suggestion
        [
          "ModalLayout",
          "NoteListBarLayout",
          "EditorLayout"
        ]
```

File: src/app/states/layouts/page.mdx [hunk 17]
```diff
@@ -0,0 +1,179 @@
...
+
+<Row>
+  <Col>
+    <Properties>
+      <Property name="main:full" type="Array<string>">
+      The main layout that consists of 3 columns.
+
+      **Default values:**
+        ```json
+        ["ModalLayout", "SideBarLayout", "NoteListBarLayout", "EditorLayout"]
```
Comment (Keisir) [inline src/app/states/layouts/page.mdx:28]:
```suggestion
        [
          "ModalLayout",
          "SideBarLayout",
          "NoteListBarLayout",
          "EditorLayout"
        ]
```

File: src/app/states/layouts/page.mdx [hunk 18]
```diff
@@ -0,0 +1,179 @@
...
+You can obtain the `layouts` state and integrate it with React components.{{ className: 'lead' }}
+
+
+---
+
+## Data Structure
+A map of `Record<string, LayoutComponents>` where:
+
+- `layout_name`: String
+- `LayoutComponents`: A class name of React component
```
Comment (Keisir) [inline src/app/states/layouts/page.mdx:18]:
```suggestion
- `string`: The layout name
- `LayoutComponents`: An array of React component class names
```

**Review by Keisir — COMMENTED** [review]
Nice work!

I made some suggestions to improve the overall readability.

**Review by Keisir — APPROVED** [review]
LGTM

**Comment by Keisir** [comment]
@MayuBen1027, a quick tip:
You can add suggestions to a batch and apply them with a single commit.
This can be done in the "Files changed" tab:
grafik
grafik

**Comment by craftzdog** [comment]
Improved the readability by making a section 'Available layouts' https://github.com/inkdropapp/api-docs/commit/2740617a9b454f328a3df7b55d4250375d8d8088

**Pull request merged**